CHANGELOG.md

```diff
@@ -5,6 +5,18 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Fixed
+
+- Improvements to the Raft implementation, to retain commit safety and liveness despite message loss (#6016).
+
+### Added
+
+- Added 2 new log lines which may be helpful diagnostics in production deployments, both including a new `[rollback]` tag.
+  - `[rollback] ... Dropping conflicting branch` may be emitted after network partitions, and indicates that some unstable (non-committed) transactions have been lost. This is expected, but worth investigating if it occurs regularly - it is a sign of elections impacting service availability.
+  - `[rollback] ... Ignoring conflicting AppendEntries` could also be emitted after a network partition, but **should be reported to the CCF development team**. It is a sign of an unexpected execution path, which could lead to loss of liveness (inability to advance commit).
```

Comment:
We may want to highlight this even more loudly. This path is retained only because we don't currently have a counter-example for it, so removing it is a larger (more dangerous) change. But if this is ever seen in the wild, we want to know and want to remove this path (likely removing the `is_new_follower` logic entirely).

Reply:
@andpiccione as discussed earlier this week.